synapse/module_api/__init__.py

```diff
@@ -1226,6 +1228,38 @@ async def sleep(self, seconds: float) -> None:
 
         await self._clock.sleep(seconds)
 
+    async def send_http_push_notification(
+        self,
+        user_id: str,
+        device_id: str,
+        content: JsonDict,
+        tweaks: Dict[str, str] = {},
+    ) -> bool:
+        """Send an HTTP push notification that is forwarded to the registered push gateway
+        for the specified device.
+
+        Added in Synapse v1.82.0.
+
+        Args:
+            user_id: The user ID of the device where to send the push notification.
+            device_id: The device ID of the device where to send the push notification.
+            content: A dict of values that will be put in the `notification` field of the push
+            (cf Push Gatway spec). `devices` field will be overrided if included.
+            tweaks: A dict of `tweaks` that will be inserted in the `devices` section, cf spec.
+
+        Returns:
+            True if at least one push was succesfully sent, False in case of error or if no
```

Comment:
I think we probably want to distinguish the 2 types of error here (no pusher vs fail) ?
Or do we want to just ignore if there is no pusher ?

I am thinking about returning false for the no pusher case, and raising an exception for the failure.